clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -406,6 +406,10 @@ public void close() {
         } finally {
             super.close();
         }
+        Node coordinator;
+        while ((coordinator = coordinator()) != null && client.pendingRequestCount(coordinator) > 0) {
+            client.poll(Long.MAX_VALUE);
```

Comment:
@hachikuji There is a `disableWakeups` a few lines above at the beginning of `close()`. Doesn't that take care of wakeups?